src/main/java/org/example/Person.java

```diff
@@ -127,18 +136,55 @@ public void addSibling(Person person){
             this.siblings.add(person);
         }
 
-        public List<Person> getGrandChildren(Person person){
-            List<Person> grandChildrens = new ArrayList<>();
-            for (Person p: person.children){
+        public List<Person> getGrandChildren(){
+            List<Person> grandChildren = new ArrayList<>();
+            for (Person p: this.children){
                 for (Person gC : p.getChildren()){
-                    grandChildrens.add(gC);
+                    grandChildren.add(gC);
                 }
             }
-            return grandChildrens;
+            return grandChildren;
         }
 
 
         public void addPet(Pet pet) {
         this.pets.add(pet);
         }
+    public void  addPartner(Person partner){
+        this.partner = partner;
+    }
+    public List<Pet> getGrandChildrensPets(){
+        List<Pet> pets = new ArrayList<>();
+        List<Person> grandChildrens = getGrandChildren();
+        for (Person gC : grandChildrens) {
+            for (Pet p : gC.pets)
+                pets.add(p);
+        }
+        return pets;
+    }
+    public List<Person> getFemaleCousins(){
+    List<Person> cousins = new ArrayList<>();
+    List<Person> motherSiblings = this.mother.getSiblings();
+    List<Person> fatherSiblings = this.father.getSiblings();
+    if(motherSiblings != null) {
+        for (Person sib : motherSiblings) {
+            for (Person c : sib.children) {
+                if (c.getSex().equals("female")) {
+                    cousins.add(c);
+                }
+            }
+        }
+    }
+    if(fatherSiblings != null) {
+        for (Person sib : fatherSiblings) {
+            for (Person c : sib.children) {
+                if (c.getSex().equals("female")) {
+                    cousins.add(c);
+                }
+            }
+        }
+    }
+
+        return cousins;
+    }
```

Comment:
Knap dat je aan de bonus bent toegekomen. Ik kwam daar zelf niet uit, en de uitwerkingen van de bonus staan niet op Github en werden ook niet besproken in de huiswerkklas, dus hier kan ik je niet bij helpen. 

Als je je branche mergt met je main bestand, kan ik hem wel van Github clonen. Dan zou ik er op die manier nog wel naar kunnen kijken in IntelliJ, als je dat interressant vindt.

Reply:
Zeker wel:)